packages/calcite-components/src/components/meter/meter.tsx

```diff
@@ -0,0 +1,522 @@
+      this.determineValueLabelPosition();
+    }
+    if (this.rangeLabels) {
+      this.determineVisibleLabels();
+    }
+  };
+
+  private calculateValues() {
+    const { low, high, min, max, value } = this;
+    const lowPercent = (100 * (low - min)) / (max - min);
+    const highPercent = (100 * (high - min)) / (max - min);
+    const currentPercent = (100 * (value - min)) / (max - min);
+    this.lowPercent = lowPercent;
+    this.highPercent = highPercent;
+    this.currentPercent = value ? currentPercent : 0;
+    this.lowActive = !!low && low > min && (!value || low > value) && (!high || low < high);
+    this.highActive = !!high && high < max && (!value || high > value) && (!low || high > low);
+  }
+
+  /**
+   * Returns a string representing the localized label.
+   *
+   * @param value
+   * @param type
+   * @param percent
+   * @returns
+   */
+
+  private formatLabel = (value: number, labelType: MeterLabelType): string => {
+    if (typeof value === "number" && labelType !== "percent") {
+      numberStringFormatter.numberFormatOptions = {
+        locale: this.effectiveLocale,
+        numberingSystem: this.numberingSystem,
+        useGrouping: this.groupSeparator,
+      };
+      return numberStringFormatter.localize(value.toString());
+    } else if (typeof value === "number" && labelType === "percent") {
+      return Intl.NumberFormat(this.effectiveLocale, {
+        useGrouping: this.groupSeparator,
+        style: "percent",
+      }).format(value);
+    }
+  };
+
+  private getMeterKind(): string {
+    const { low, high, min, max, value } = this;
+    const lowest = low ? low : min;
+    const highest = high ? high : max;
+    const aboveLowest = value >= lowest;
+    const belowLowest = value < lowest;
+    const aboveHighest = value >= highest;
+    const belowHighest = value < highest;
+
+    if (!value || (!low && belowHighest) || belowLowest) {
+      return CSS.success;
+    } else if (aboveLowest && belowHighest) {
+      return CSS.warning;
+    } else if (aboveHighest) {
+      return CSS.danger;
+    }
+  }
+
+  private comparePosition(el1: HTMLDivElement, el2: HTMLDivElement) {
+    if (el1 && el2) {
+      return intersects(el1?.getBoundingClientRect(), el2?.getBoundingClientRect());
+    }
+  }
+
+  private determineVisibleLabels() {
+    const { minLabelEl, lowLabelEl, highLabelEl, maxLabelEl } = this;
+    const minLowOverlap = this.comparePosition(minLabelEl, lowLabelEl);
+    const minHighOverlap = this.comparePosition(minLabelEl, highLabelEl);
+    const minMaxOverlap = this.comparePosition(minLabelEl, maxLabelEl);
+    const lowMaxOverlap = this.comparePosition(lowLabelEl, maxLabelEl);
+    const lowHighOverlap = this.comparePosition(lowLabelEl, highLabelEl);
+    const highMaxOverlap = this.comparePosition(highLabelEl, maxLabelEl);
+
+    if (lowLabelEl) {
+      const hideLowLabel = minLowOverlap || lowMaxOverlap || lowHighOverlap;
+      lowLabelEl.style.opacity = hideLowLabel ? "0" : "100";
```

Comment:
Well it WAS because I was encountering issues using Tailwind's `@apply hidden` in a conditionally applied class. But taking another look, I see that I've been unintentionally setting `display: none` and not `visibility: hidden` - which Tailwind of course calls `invisible` instead 😄 

With it toggling `display: none` instead - it was messing with all the pixel width calculations we do for overlap. Of course - using the correct TW name makes it work fine, lol.

Reply:
The other locations where styles are set inline are where we calculate position and set a width or position value, so I don't know if those would work in classes.